BackofficeBundle/EventSubscriber/NodeTemplateSelectionSubscriber.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace OpenOrchestra\BackofficeBundle\EventSubscriber;
+
+use OpenOrchestra\BackofficeBundle\Manager\NodeManager;
+use OpenOrchestra\ModelInterface\Repository\TemplateRepositoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * Class NodeTemplateSelectionSubscriber
+ */
+class NodeTemplateSelectionSubscriber implements EventSubscriberInterface
+{
+    protected $nodeManager;
+    protected $templateRepository;
+
+    /**
+     * @param NodeManager                 $nodeManager
+     * @param TemplateRepositoryInterface $templateRepository
+     */
+    public function __construct(NodeManager $nodeManager, TemplateRepositoryInterface $templateRepository)
+    {
+        $this->nodeManager = $nodeManager;
+        $this->templateRepository = $templateRepository;
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSubmit(FormEvent $event)
+    {
+        $form = $event->getForm();
+        $formData = $form->getData();
+        $data = $event->getData();
+
+        if (
+            array_key_exists('nodeTemplateSelection', $data) &&
+            array_key_exists('templateId', $data['nodeTemplateSelection']) &&
+            null === $formData->getId() &&
+            0 === $formData->getAreas()->count() &&
+            0 === $formData->getBlocks()->count()
+        ) {
+            $template = $this->templateRepository->findOneByTemplateId($data['nodeTemplateSelection']['templateId']);
+            if (null !== $template) {
+                $formData->setAreas($template->getAreas());
+                $formData->setBlocks($template->getBlocks());
+            }
+        }
+
+        if (
```

Comment:
I would rather put an `elseif` instead of only a `if`.

I think that both behaviour cannot be done at the same time